FirebaseAdmin/FirebaseAdmin/AppOptions.cs

```diff
@@ -25,6 +26,8 @@ namespace FirebaseAdmin
     /// </summary>
     public sealed class AppOptions
     {
+        private HttpClientFactory httpClientFactory;
```

Comment:
Can we do this instead?

```
public AppOptions()
{
  this.HttpClientFactory = new HttpClientFactory();
}

public HttpClientFactory HttpClientFactory { get; set; }
```

Then in `FirebaseApp` constructor:

```
if (this.options.HttpClientFactory == null)
{
  // throw
}
```

Without an explicit throw, we could accidentally mask an error in user code.